teamcity: failed test: TestTxnWaitQueueEnableDisable

The following tests appear to have failed on master (testrace): TestTxnWaitQueueEnableDisable

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestTxnWaitQueueEnableDisable).

[#1281492](https://teamcity.cockroachdb.com/viewLog.html?buildId=1281492):


```
TestTxnWaitQueueEnableDisable
--- FAIL: testrace/TestTxnWaitQueueEnableDisable (0.100s)

------- Stdout: -------
I190508 22:59:29.176245 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190508 22:59:29.176615 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
1      [async] intent_resolver_ir_batcher
I190508 22:59:29.176767 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
    txn_wait_queue_test.go:136: expected QueryWaiting to be 0, got -1 instead
------- Stdout: -------
I190508 22:47:10.662774 38525 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190508 22:47:10.662836 38525 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher



TestTxnWaitQueueEnableDisable
--- FAIL: testrace/TestTxnWaitQueueEnableDisable (0.060s)

------- Stdout: -------
I190508 22:59:29.176245 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190508 22:59:29.176615 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
1      [async] intent_resolver_ir_batcher
I190508 22:59:29.176767 40304 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] monitoring pusher txn
    txn_wait_queue_test.go:136: expected QueryWaiting to be 0, got -1 instead




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/txnwait/txnqueue.go
+++ b/pkg/storage/txnwait/txnqueue.go
@@ -644,7 +644,7 @@ func (q *Queue) MaybeWaitForQuery(
 	if !req.WaitForUpdate {
 		return nil
 	}
-
+	metrics := q.store.GetTxnWaitMetrics()
 	q.mu.Lock()
 	// If the txn wait queue is not enabled or if the request is not
 	// contained within the replica, do nothing. The request can fall
@@ -684,10 +684,9 @@ func (q *Queue) MaybeWaitForQuery(
 		}
 		q.mu.queries[req.Txn.ID] = query
 	}
+	metrics.QueryWaiting.Inc(1)
 	q.mu.Unlock()
 
-	metrics := q.store.GetTxnWaitMetrics()
-	metrics.QueryWaiting.Inc(1)
 	tBegin := timeutil.Now()
 	defer func() { metrics.QueryWaitTime.RecordValue(timeutil.Since(tBegin).Nanoseconds()) }()
 

